src/index.ts

```diff
@@ -34,6 +34,12 @@ class Gtmhub {
     window.addEventListener("message", (event) => {
       const { type, data } = event.data;
 
+      // messages can come from different origins, so make sure
+      // we use only expected ones
+      if (!type) {
```

Comment:
Guess we can strengthen this check with explicitly checking the event.origin, could be something as follows.
`if (!event.origin.includes(".gtmhub.com")) { return }`

Reply:
I wanted to use origin in the beginning. But since we deploy some apps in *.[gtmhub.com](http://gtmhub.com/) it won’t work. The origin should be the same as the website where it is used, e.g. [live.gtmhub.com](http://live.gtmhub.com/), but I don’t know this value. Maybe we could inject this URL when initializing the plugin, since the webapp knows its URL